### ISSUE TITLE
🐛 [Bug Report] mangos-classic  compile error(AI)

### ISSUE THREAD

**Issue by bibibi-bug** [cmangos/issues#2321]
### Version & Environment
  - "1.12.1" (CLASSIC)

### The wrong location
```
In file included from /home/mangos/cmangos/src/game/AI/BaseAI/CreatureAI.h:22:0,
                 from /home/mangos/cmangos/src/game/AI/BaseAI/CreatureAI.cpp:19:
/home/mangos/cmangos/src/game/AI/BaseAI/UnitAI.h:462:39: error: 'std::function' has not been declared
         void AttackSpecificEnemy(std::function<void(Unit*,Unit*&)> check);
                                       ^~~~~~~~
/home/mangos/cmangos/src/game/AI/BaseAI/UnitAI.h:462:47: error: expected ',' or '...' before '<' token
         void AttackSpecificEnemy(std::function<void(Unit*,Unit*&)> check);
                                               ^
src/game/CMakeFiles/game.dir/build.make:62: recipe for target 'src/game/CMakeFiles/game.dir/AI/BaseAI/CreatureAI.cpp.o' failed
make[2]: *** [src/game/CMakeFiles/game.dir/AI/BaseAI/CreatureAI.cpp.o] Error 1
CMakeFiles/Makefile2:506: recipe for target 'src/game/CMakeFiles/game.dir/all' failed
make[1]: *** [src/game/CMakeFiles/game.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 19%] Building CXX object src/realmd/CMakeFiles/realmd.dir/RealmList.cpp.o
[ 19%] Linking CXX executable realmd
make: *** [all] Error 2
[ 19%] Built target realmd
Makefile:129: recipe for target 'all' failed
```

**Comment by bibibi-bug** [comment]
The latest compilation
commit 2d3e7e5bc4b7b14cd41ca3c2e6e6147d7c2fb3fa

**Comment by bibibi-bug** [comment]
[cmangos/mangos-tbc@c4cf8d1](https://github.com/cmangos/mangos-tbc/commit/c4cf8d1191c7fb396e9fe1223ee5808ae9bc0cf8)
According to use can compile successfully
Looking forward to update COMMIT

**Comment by jimmybrancaccio** [comment]
Hello @bibibi-bug Please go back and fill out _all_ of the informed requested by the bug reporting templated. Once that has been done we can move forward. Thanks!

**Comment by evil-at-wow** [comment]
It's the same issue as in #2322 but for Classic. We also got the same issue in the TBC core, see #2304. I've fixed it there in cmangos/mangos-tbc@c4cf8d1 but it looks like killerwife skipped my commits when porting the most recent block of commits in the TBC core to Classic.

**Comment by bibibi-bug** [comment]
@evil-at-wow yes
